src/scene/gsplat/gsplat.js

```diff
@@ -120,12 +130,18 @@ class GSplat {
         result.setParameter('transformA', this.transformATexture);
         result.setParameter('transformB', this.transformBTexture);
         result.setParameter('numSplats', this.numSplatsVisible);
-        if (this.hasSH) {
-            result.setDefine('SH_BANDS', 3);
+        if (this.shBands > 0) {
+            result.setDefine('SH_BANDS', this.shBands);
             result.setParameter('splatSH_1to3', this.sh1to3Texture);
-            result.setParameter('splatSH_4to7', this.sh4to7Texture);
-            result.setParameter('splatSH_8to11', this.sh8to11Texture);
-            result.setParameter('splatSH_12to15', this.sh12to15Texture);
+            if (this.shBands > 1) {
```

Comment:
it'd be simpler to keep the original code and simply let it assign parameters as null or so

Reply:
I would have except 2 bands uses `splatSH_8` and 3 bands uses `splatSH_8to11`...

Reply:
I mean, maybe that's fine too